src/Selenium2Driver.php

```diff
@@ -952,10 +952,10 @@ public function wait($timeout, $condition)
         $start = microtime(true);
         $end = $start + $timeout / 1000.0;
 
-        do {
-            $result = $this->wdSession->execute(array('script' => $script, 'args' => array()));
+        while (microtime(true) < $end && !$this->wdSession->execute(array('script' => $script, 'args' => array())))
```

Comment:
the only case I can imagine in which it would not run at all is when `microtime(true) < $end` in the very first iteration of the loop. That would happen if the user specify a timeout that is shorter than the time needed for setting up the while loop and test `microtime(true) < $end` but that is fine because in that case the timeout is actually run up.

But I also could reverse the loop
`while (!$this->wdSession->execute(array('script' => $script, 'args' => array())) && microtime(true) < $end)`
Then the JS will run always at least once, no matter of the timeout